pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -216,3 +217,115 @@ func validateExtendedResourceName(name v1.ResourceName) []error {
 	}
 	return validationErrors
 }
+
+// ValidateInterPodAffinityArgs validates that InterPodAffinityArgs are correct.
+func ValidateInterPodAffinityArgs(args config.InterPodAffinityArgs) error {
+	return ValidateHardPodAffinityWeight(field.NewPath("hardPodAffinityWeight"), args.HardPodAffinityWeight)
+}
```

Comment:
testcases for those validation should be migrated to "pkg/scheduler/apis/config/validation/validation_test.go" and remove from the legacy plugin testcases, add new testcases if they are missed.